diff --git a/local_install/share/groff/tmac/an-old.tmac b/local_install/share/groff/tmac/an-old.tmac
index 7b14918..bea67c8 100644
--- a/local_install/share/groff/tmac/an-old.tmac
+++ b/local_install/share/groff/tmac/an-old.tmac
@@ -43,8 +43,6 @@
 .do nr *groff_an-old_tmac_C \n[.cp]
 .cp 0
 .
-.if (\n[.x]\n[.y] < 118) \
-.  ab You need GNU troff 1.18 or higher to run this version of man!
 .
 .\" Define a string for use in diagnostic messages.
 .ds an-old an-old.tmac
